src/linear_algebra/projective_space/independence.lean

```diff
@@ -0,0 +1,109 @@
+/-
+Copyright (c) 2022 Michael Blyth. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Blyth
+-/
+
+import linear_algebra.projective_space.basic
+
+/-!
+# Independence in Projective Space
+
+In this file we define independence and dependence of families of elements in projective space.
+
+## Implementation Details
+
+We use an inductive definition to define the independence of points in projective
+space, where the only constructor assumes an independent family of vectors from the
+ambient vector space. Similarly for the definition of dependence.
+
+## Results
+
+- A family of elements is dependent if and only if it is not independent.
+- Two elements are dependent if and only if they are equal.
+
+# Future Work
+
+- Define collinearity in projective space.
+- Prove the axioms of a projective geometry are satisfied by the dependence relation.
+- Define projective linear subspaces.
+-/
+
+
+variables {K V : Type*} [field K] [add_comm_group V] [module K V]
+
+namespace projectivization
+
+/-- A linearly independent familty of nonzero vectors gives an independent family of points
+in the projective space. --/
+inductive independent {ι : Type*} : (ι → ℙ K V) → Prop
```

Comment:
It's quite rare that we need to use `inductive` and at first glance it doesn't look necessary here. Is there a reason I'm missing?

If not, it looks to me like a better definition would be either:
```lean
def independent {ι : Type*} (f : ι → ℙ K V) := linear_independent K $ projectivization.rep ∘ f
```
(which of course you show is equivalent) or maybe even better:
```lean
def independent {ι : Type*} (f : ι → ℙ K V) := complete_lattice.independent $ λ i, (f i).submodule
```

Reply:
NB: it's possible #14714 might be relevant

Reply:
I think the main benefit of an inductive approach is that by using the `rcases` or `induction` tactic you automatically get representatives for your inductive family.
This also has the benefit that it's easy to construct independence proofs if you already have an independent family of vectors.

At the end of the day, this is equivalent to showing that `projectivization.rep ∘ f` is independent.

If we had an analogue of `quotient.lift_on` for families of elements of quotients as opposed to just 1, 2 or 3 elements, then I think that would be the ideal way to define independence in this case (projectivization is defined as a quotient).

Related zulip discussion:
https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/quotient.2Elift_on_family/near/286096782

All that aside, I do think we should try to include in this PR the equivalence with the lattice-theoretic definition of independence in your second code block!

Reply:
Thanks @adamtopaz , let's keep the use of `inductive` then.

Regarding the equivalence with the lattice-theoretic definition, this will follow trivially from `independent_iff_linear_independent_of_ne_zero` included in #14714